add "Decorating your data" part (#2048)
PrestaShop/docs: development/components/grid/_index.md

        - '@prestashop.core.grid.filter.form_factory'                            # core service needed by grid factory
        - '@prestashop.core.hook.dispatcher'                                     # core service needed by grid factory
```
And we are done!

#### Decorating your data

> Sometimes, you may need to decorate or modify your data in order to format it in a specific way. In this case, you can create your own decorator service.

Reviewer: ```suggestion
Sometimes you may need to modify or decorate your data to format it in a specific way. In this case, you can create a custom decorator service.

This is useful when you want to keep your data retrieval logic untouched and apply formatting, transformations, or additional computed fields before rendering the grid.
```

First, declare the decorator service in your YAML configuration:

```yaml
my.namespace.grid.data_provider.product_grid_data_factory_decorator:
    class: 'My\Namespace\Grid\Data\ProductGridDataFactoryDecorator'
    arguments:
      - '@prestashop.core.grid.data.factory.product_data_factory' # our data factory
```

> Then you have to create the corresponding class. You can alter the records in the `applyModifications` function.

Reviewer: ```suggestion
Then, create the corresponding class. You can alter the records inside the `applyModifications()` method.
```

use PrestaShop\PrestaShop\Core\Grid\Data\Factory\GridDataFactoryInterface;

final class ProductGridDataFactoryDecorator implements GridDataFactoryInterface
{
    private $productGridDataFactory;

    public function __construct(
        GridDataFactoryInterface $productGridDataFactory
    ) {
        $this->productGridDataFactory = $productGridDataFactory;
    }

    public function getData(SearchCriteriaInterface $searchCriteria)
    {
        $productData = $this->productGridDataFactory->getData($searchCriteria);

        $productRecords = $this->applyModifications($productData->getRecords());

        return new GridData(
            $productRecords,
            $productData->getRecordsTotal(),
            $productData->getQuery()
        );
    }

    private function applyModifications(RecordCollectionInterface $products)
    {
        $modifiedProducts = [];

        foreach ($products as $product) {
            # Basic example on how to alter the record
            $product['name'] = ucfirst($product['name']);

            $modifiedProducts[] = $product;
        }

        return new RecordCollection($modifiedProducts);
    }
}

```

> Finally, we override the default `Grid Factory` service to use our decorated data factory instead of the original one.

Reviewer: ```suggestion
Finally, override the default `GridFactory` service to use the decorated data factory instead of the original one.
```
